mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -580,10 +581,14 @@ class Word2VecModel private[spark] (
       ind += 1
     }
 
-    val scored = wordList.zip(cosVec).toSeq.sortBy(-_._2)
+    val pq = new BoundedPriorityQueue[(String, Double)](num + 1)(Ordering.by(_._2))
+
+    pq ++= wordList.zip(cosVec)
```

Comment:
I also don't know... when I've used it it has been with vocabs of tens of thousands of words. From others' emails I think some people do use it with very large vocabs. If you have a minute, while we're here, might as well take it one more step towards optimized?